src/algebra/z64/batch.rs

```diff
@@ -0,0 +1,105 @@
+use super::*;
+
+use crate::util::Writer;
+
+use serde::{Deserialize, Serialize};
+
+use std::convert::TryInto;
+use std::io;
+
+//Batches for Z64 are always dimension 1
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Batch(pub(super) u64);
+
+impl Add for Batch {
+    type Output = Self;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    #[inline(always)]
+    fn add(self, other: Self) -> Self::Output {
+        Self(u64::wrapping_add(self.0, other.0))
+    }
+}
+
+impl Sub for Batch {
+    type Output = Self;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    #[inline(always)]
+    fn sub(self, other: Self) -> Self::Output {
+        Self(u64::wrapping_mul(self.0, other.0))
```

Comment:
Should be wrapping_sub. To catch this kind of errors in future Domain implementations, I would say we add some basic arithmetic testing in the domain tests.